### PR TITLE
refactor: Refactoring DataSource to DiscoveryService and updating tests

### PR DESCRIPTION
**Pull request by jungwire**
### Description:

Rename and simplify datasource table to discoveryservice

### Checklist:

_Make sure you tick all the boxes bellow if they are true or do not apply:_

- [x] I have performed a self review of my code
- [x] My code follows [Google Java Code style](https://google.github.io/styleguide/javaguide.html)
- [x] I have made my code as simple as possible
- [x] I have added unit tests and the code coverage has not decreased
- [x] I have updated the documentation in all relevant places

### CODE REVIEW

File: src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import lombok.ToString;
+
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Builder
+@Entity(name = "DiscoveryService")
+@Table(name = "discovery_service")
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java:25]: not sure if these are needed

File: src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    DiscoveryService that = (DiscoveryService) o;
+    return Objects.equals(getName(), that.getName()) && Objects.equals(getUrl(), that.getUrl());
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(getName(), getUrl());
+  }
+
+  public enum ApiType {
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java:56]: is this necessary?

File: src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+@ToString
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Builder
+@Entity(name = "DiscoveryService")
+@Table(name = "discovery_service")
+public class DiscoveryService extends AuditEntity {
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java:26]: add javadoc

**Review by RadovanTomik — CHANGES_REQUESTED** [review]

File: src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java (hunk 1 above)
Comment (jungwire) [reply src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java]: removed @Table

File: src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java (hunk 2 above)
Comment (jungwire) [reply src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java]: removed ApiType

File: src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java (hunk 3 above)
Comment (jungwire) [reply src/main/java/eu/bbmri_eric/negotiator/database/model/DiscoveryService.java:25]: added javadoc

File: src/main/java/eu/bbmri_eric/negotiator/api/controller/v3/DiscoverServiceController.java [hunk 4]
```diff
@@ -23,37 +23,38 @@
...
   @ResponseStatus(HttpStatus.NO_CONTENT)
-  DataSourceDTO update(
-      @PathVariable Long id, @Validated(Update.class) @RequestBody DataSourceCreateDTO request) {
-    return dataSourceService.update(id, request);
+  DiscoveryServiceDTO update(
+      @PathVariable Long id,
+      @Validated(Update.class) @RequestBody DiscoveryServiceCreateDTO request) {
+    return discoveryServiceService.update(id, request);
   }
 }
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/api/controller/v3/DiscoverServiceController.java:60]: Please change to /discovery-services, see https://restfulapi.net/resource-naming/

File: src/main/java/eu/bbmri_eric/negotiator/configuration/security/OAuthSecurityConfig.java [hunk 5]
```diff
@@ -122,9 +122,9 @@ public SecurityFilterChain filterChain(HttpSecurity http, MvcRequestMatcher.Buil
                     .authenticated()
                     .requestMatchers(mvc.pattern(HttpMethod.PUT, "/v3/requests/**"))
                     .authenticated()
-                    .requestMatchers(mvc.pattern(HttpMethod.POST, "/v3/data-sources/**"))
+                    .requestMatchers(mvc.pattern(HttpMethod.POST, "/v3/discovery-service/**"))
                     .hasAuthority("ADMIN")
-                    .requestMatchers(mvc.pattern(HttpMethod.PUT, "/v3/data-sources/**"))
+                    .requestMatchers(mvc.pattern(HttpMethod.PUT, "/v3/discovery-service/**"))
                     .hasAuthority("ADMIN")
```
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/configuration/security/OAuthSecurityConfig.java:128]: see comment above

File: src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/DiscoveryServiceDTO.java
Comment (RadovanTomik) [inline src/main/java/eu/bbmri_eric/negotiator/dto/discoveryservice/DiscoveryServiceDTO.java:1]: the DTO has an attribute description but the enditity doesn't have one, so please either add it to entity or remove it from DTO.

**Review by RadovanTomik — CHANGES_REQUESTED** [review]

**Review by RadovanTomik — APPROVED** [review]

**Pull request merged**